TPKeyboardAvoiding/UIScrollView+TPKeyboardAvoidingAdditions.m

```diff
@@ -426,7 +426,7 @@ -(CGFloat)TPKeyboardAvoiding_idealOffsetForView:(UIView *)view withViewingAreaHe
             CGRect cursorRect = [textInput caretRectForPosition:textInput.selectedTextRange.start];
             cursorRect = [view convertRect:cursorRect toView:self];
             CGRect visibleRect = self.bounds;
-            visibleRect.size.height = viewAreaHeight;
+            visibleRect.size.height = viewAreaHeight - self.keyboardAvoidingPadding;
```

Comment:
We need to take into account the amount the offset may be padded in the visible rect, otherwise the cursor may be below the area where we want it to be.